### ISSUE TITLE
Cannot reference Type variable defined in class block

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#886]
**Describe the bug**
Variables in the form of `Type[Foo]` cannot be used in annotations if attached to a class, though explicit type aliases can.

**To Reproduce**
Type check the snippet beloww.

**Expected behavior**
That `ClassBar.bar` would be a valid annotation for a variable of type `str`.

**Screenshots or Code**

```py
# pyright: strict

from __future__ import annotations

from typing import Type


Foo = str
Bar: Type[str]

foo: Foo    # Ok
bar: Bar    # Ok


class ClassFoo:
    foo = str

class ClassBar:
    bar: Type[str]

class_foo: ClassFoo.foo     # Ok
class_bar: ClassBar.bar     # 'Expected class type but received "Type[str]"'

```

**VS Code extension or command-line**
Extension, 1.1.55

**Comment by erictraut** [comment]
What you're attempting to do here isn't really supported. The fact that the first example `Bar: Type[str]` works is arguably a bug. Type annotations should refer to symbols that have assigned values, not unbound symbols.

So both of your examples involving the symbol `bar` are incorrect. The examples involving `foo` are correct.

**Comment by layday** [comment]
This doesn't work either however:

```py
# pyright: strict

from typing import Type

class ClassBar:
    bar: Type[str] = str

# Expected class type but received "Type[str]"
# Expression of type "Literal['']" cannot be assigned to declared type "Type[str]"
#  Type "Literal['']" cannot be assigned to type "Type[str]"
class_bar: ClassBar.bar = ''

```

**Comment by erictraut** [comment]
That's a legit bug.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.56, which I just published. It will also be in the next version of Pylance.
